qa/L0_libtorch_io_names/io_names_client.py

```diff
@@ -0,0 +1,95 @@
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+sys.path.append("../common")
+
+from builtins import range
+from future.utils import iteritems
+import unittest
+import test_util as tu
+import numpy as np
+
+import tritonclient.http as httpclient
+from tritonclient.utils import np_to_triton_dtype
+from tritonclient.utils import InferenceServerException
+
+
+class IONamingConvention(tu.TestResultCollector):
+
+    def _infer_helper(self, model_name, io_names):
+        triton_client = httpclient.InferenceServerClient("localhost:8000", verbose=False)
+
+        # Create the data for the two inputs. Initialize the first to unique
+        # integers and the second to all ones.
+        input0_data = np.arange(start=0, stop=16, dtype=np.float32)
+        input0_data = np.expand_dims(input0_data, axis=0)
+        input1_data = np.full(shape=(1, 16), fill_value=-1, dtype=np.float32)
+
+        inputs = []
+        output_req = []
+        inputs.append(
+            httpclient.InferInput(io_names[0], [1, 16], "FP32"))
+        inputs[-1].set_data_from_numpy(input0_data)
+        inputs.append(
+            httpclient.InferInput(io_names[1], [1, 16], "FP32"))
+        inputs[-1].set_data_from_numpy(input1_data)
+        output_req.append(
+            httpclient.InferRequestedOutput(io_names[2],
+                                            binary_data=True))
+        output_req.append(
+            httpclient.InferRequestedOutput(io_names[3],
+                                            binary_data=True))
+
+        results = triton_client.infer(model_name,
+                                    inputs,
+                                    outputs=output_req)
+
+        output0_data = results.as_numpy(io_names[2])
+        output1_data = results.as_numpy(io_names[3])
+        for i in range(16):
+            self.assertEqual(input0_data[0][i] - input1_data[0][i], output0_data[0][i])
+            self.assertEqual(input0_data[0][i] + input1_data[0][i], output1_data[0][i])
+
+    def test_io_index(self):
+        io_names = ["INPUT__0", "INPUT__1", "OUTPUT__0", "OUTPUT__1"]
+        self._infer_helper("libtorch_io_index", io_names)
+
+    def test_output_index(self):
+        io_names = ["INPUT0", "INPUT1", "OUTPUT__0", "OUTPUT__1"]
+        self._infer_helper("libtorch_output_index", io_names)
+
+    def test_no_output_index(self):
+        io_names = ["INPUT0", "INPUT1", "OUTPUT0", "OUTPUT1"]
+        self._infer_helper("libtorch_no_output_index", io_names)
+
+    def test_no_arguements_no_output_index(self):
+        io_names = ["INPUTA", "INPUTB", "OUTPUTA", "OUTPUTB"]
+        self._infer_helper("libtorch_no_arguements_output_index", io_names)
```

Comment:
Add test for the mix of different rules.

Reply:
Added the mix of rules as well

1. ["INPUTA", "INPUT__1", "OUTPUTA", "OUTPUT__1"]
2. ["INPUT0", "INPUTB", "OUTPUTA", "OUTPUT__1"]
3. ["INPUT0", "INPUT__1", "OUTPUT0", "OUTPUT__1"]
4.  ["INPUT1", "INPUT0", "OUTPUT__1", "OUTPUT__0"] (unordered)